data-8/jupyterhub-k8s
# Adding nbgdrive and fixing nbinteract

Summary
-------
Adds nbgdrive to the single user notebook. Adds a fix for nbinteract so that Interact links do not break when a repository is dirty.

How to test
-------
To test nbgdrive:
1. Login and create a new notebook.
2. Once the notebook is open, check the top right corner on the toolbar to see if there is a "Verify Drive" button exists.
3. Press the "Verify Drive" button.
4. Copy the code google gives you on the new tab the button opens and paste it in the newly created text box where the "Verify Drive" button was.
5. Click the button next to the textbox to authorize.
6. After verification, nbgdrive should do an initial sync. You can check your google drive associated with the google account you signed in for during verification to confirm that it worked.
7. Try creating other files in your jupyter notebook server session and press the small icon resembling a cloud in the toolbar in the notebook interface. If you hover over the button with your mouse, the help text should say something like "Manually Sync..." etc.
8. After pressing the button, nothing apparent will happen. This is normal. Perhaps this needs to change in the future so users know that the button is not broken. However, you can right click and "inspect element" on chrome. Then, select the "console" tag to wait for a javascript log saying that the syncing has finished.
9. You can check your google drive to confirm that the sync worked.

To test the interact fix:
1. Login to a single user server session.
2. Pull in a repo using an Interact link eg. http://datahub-dev.berkeley.edu/hub/user-redirect/interact?repo=textbook&path=notebooks/Types.ipynb
3. Edit a file in the repo and save the changes you made.
4. Try navigating to the same Interact link as before.
5. The Interact link should not error out. If a path was specified, it should successfully redirect you to that path in the repo.
